sql/stats: TestCreateStatsControlJob failed

sql/stats.TestCreateStatsControlJob [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3596916&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3596916&tab=artifacts#/) on master @ [5d972a683c531326bc0af403c6d3373d3f4b2267](https://github.com/cockroachdb/cockroach/commits/5d972a683c531326bc0af403c6d3373d3f4b2267):


```
=== RUN   TestCreateStatsControlJob
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCreateStatsControlJob598461668
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCreateStatsControlJob
    create_stats_job_test.go:125: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCreateStatsControlJob598461668
--- FAIL: TestCreateStatsControlJob (24.44s)
=== RUN   TestCreateStatsControlJob/pause
    create_stats_job_test.go:113: pq: job with status pause-requested cannot be resumed
    --- FAIL: TestCreateStatsControlJob/pause (1.90s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestCreateStatsControlJob PKG=./pkg/sql/stats TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
/cc @cockroachdb/sql-queries rytaft
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCreateStatsControlJob.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/stats.TestCreateStatsControlJob [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3705310&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3705310&tab=artifacts#/) on master @ [2de17e7fbe66e14039fc7969a76139625761438f](https://github.com/cockroachdb/cockroach/commits/2de17e7fbe66e14039fc7969a76139625761438f):


```
=== RUN   TestCreateStatsControlJob
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCreateStatsControlJob667078793
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCreateStatsControlJob
    create_stats_job_test.go:125: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCreateStatsControlJob667078793
--- FAIL: TestCreateStatsControlJob (32.37s)
=== RUN   TestCreateStatsControlJob/pause
    create_stats_job_test.go:113: pq: job with status pause-requested cannot be resumed
    --- FAIL: TestCreateStatsControlJob/pause (2.12s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCreateStatsControlJob.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/stats.TestCreateStatsControlJob [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3957342&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3957342&tab=artifacts#/) on master @ [93c9a3bc0e234c23df45aee62f6f76eb3b5ab723](https://github.com/cockroachdb/cockroach/commits/93c9a3bc0e234c23df45aee62f6f76eb3b5ab723):


```
=== RUN   TestCreateStatsControlJob
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCreateStatsControlJob840138595
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCreateStatsControlJob
    create_stats_job_test.go:130: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCreateStatsControlJob840138595
--- FAIL: TestCreateStatsControlJob (27.39s)
=== RUN   TestCreateStatsControlJob/pause
    create_stats_job_test.go:118: pq: job with status pause-requested cannot be resumed
    --- FAIL: TestCreateStatsControlJob/pause (2.42s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCreateStatsControlJob.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/stats.TestCreateStatsControlJob [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4115753&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4115753&tab=artifacts#/) on master @ [d6b99e92bf55b6f4a0d79800d67924e04d0b2a6d](https://github.com/cockroachdb/cockroach/commits/d6b99e92bf55b6f4a0d79800d67924e04d0b2a6d):


```
=== RUN   TestCreateStatsControlJob
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCreateStatsControlJob3135015947
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCreateStatsControlJob
    create_stats_job_test.go:130: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCreateStatsControlJob3135015947
--- FAIL: TestCreateStatsControlJob (30.76s)
=== RUN   TestCreateStatsControlJob/pause
    create_stats_job_test.go:118: pq: job 727706120704720897: job with status pause-requested cannot be resumed
    --- FAIL: TestCreateStatsControlJob/pause (2.34s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCreateStatsControlJob.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/stats.TestCreateStatsControlJob [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4151277&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4151277&tab=artifacts#/) on master @ [d1eafdf56b42ca09a169c199916917a144c2c412](https://github.com/cockroachdb/cockroach/commits/d1eafdf56b42ca09a169c199916917a144c2c412):


```
=== RUN   TestCreateStatsControlJob
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCreateStatsControlJob2559815458
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCreateStatsControlJob
    create_stats_job_test.go:130: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCreateStatsControlJob2559815458
--- FAIL: TestCreateStatsControlJob (29.23s)
=== RUN   TestCreateStatsControlJob/pause
    create_stats_job_test.go:118: pq: job 729150949192892417: job with status pause-requested cannot be resumed
    --- FAIL: TestCreateStatsControlJob/pause (2.36s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCreateStatsControlJob.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/stats.TestCreateStatsControlJob [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4160995&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4160995&tab=artifacts#/) on master @ [9b0065ca2450205a34d4237be7b317c2d895658c](https://github.com/cockroachdb/cockroach/commits/9b0065ca2450205a34d4237be7b317c2d895658c):


```
=== RUN   TestCreateStatsControlJob
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCreateStatsControlJob4284490729
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCreateStatsControlJob
    create_stats_job_test.go:130: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCreateStatsControlJob4284490729
--- FAIL: TestCreateStatsControlJob (33.39s)
=== RUN   TestCreateStatsControlJob/pause
    create_stats_job_test.go:118: pq: job 729417993281863681: job with status pause-requested cannot be resumed
    --- FAIL: TestCreateStatsControlJob/pause (2.94s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCreateStatsControlJob.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/stats/create_stats_job_test.go
+++ b/pkg/sql/stats/create_stats_job_test.go
@@ -106,18 +106,11 @@ func TestCreateStatsControlJob(t *testing.T) {
 		sqlDB.CheckQueryResults(t,
 			`SELECT statistics_name, column_names, row_count FROM [SHOW STATISTICS FOR TABLE d.t]`,
 			[][]string{})
-		opts := retry.Options{
-			InitialBackoff: 1 * time.Millisecond,
-			MaxBackoff:     time.Second,
-			Multiplier:     2,
-		}
-		if err := retry.WithMaxAttempts(context.Background(), opts, 10, func() error {
+
+		testutils.SucceedsSoon(t, func() error {
 			_, err := sqlDB.DB.ExecContext(context.Background(), `RESUME JOB $1`, jobID)
 			return err
-		}); err != nil {
-			t.Fatal(err)
-		}
-
+		})
 		jobutils.WaitForJob(t, sqlDB, jobID)
 
 		// Now the job should have succeeded in producing stats.

